### PR TITLE
API User - Get and update profile data

### PR DESCRIPTION
**Pull request by BElluu**
null

### CODE REVIEW

File: packages/backEnd/src/services/user.services.ts [hunk 1]
```diff
@@ -14,6 +14,18 @@ type User = {
...
+  address: string;
+  city: string;
+  state: string;
+  postCode: string;
+  country: string;
+  website: string;
+  phone: string;
+  birthDay: Date;
+  image: Binary
 };
```
Comment (Bartek-Figat) [inline packages/backEnd/src/services/user.services.ts:29]: I think to move this type  User to the interface folder

File: packages/backEnd/src/services/user.services.ts [hunk 2]
```diff
@@ -126,4 +138,63 @@ export class UserService {
...
+      token: {
+        token: string;
+      };
+    };
+
+    try {
+      const user = await this.repository.findOne(
+        { _id: new ObjectId(token.token) },
+        { firstName: 1, lastName: 1, company: 1, address: 1, city: 1,state: 1,
+          postCode: 1, country: 1, website: 1, phone: 1, birthDay: 1, image: 1, _id: 0 }
```
Comment (Bartek-Figat) [inline packages/backEnd/src/services/user.services.ts:153]: I can apply here _id: 0

File: packages/backEnd/src/tools/image.ts [hunk 3]
```diff
@@ -12,17 +12,19 @@ const s3 = new S3({
...
-      return Location
-    };
-  
\ No newline at end of file
+const fileToUpload = {
+    Bucket: bucketName,
+    Key: `bbardsImages/${uuidv4() + "-extension" + file.originalName + "." + extension}`,
+    Body: data,
+    ACL: "public-read",
+    ContentEncoding: 'base64',
```
Comment (Bartek-Figat) [inline packages/backEnd/src/tools/image.ts:25]: I think you don't need to use **base64** you can just specify ContentType and this should do the work.

File: packages/backEnd/src/validation/index.ts [hunk 4]
```diff
@@ -1,6 +1,7 @@
 import { RequestHandler } from "express";
 import validate from "validate.js";
 
+var unallowedCharacters = new RegExp("^.*?(?=[\^#%&$\*:<>\?/\{\|\}]).*$")
```
Comment (Bartek-Figat) [inline packages/backEnd/src/validation/index.ts:4]: What is the purpose using regex here?

File: packages/backEnd/src/services/user.services.ts [hunk 5]
```diff
@@ -152,12 +152,8 @@ export class UserService {
       };
     };
     try{
-    const user = await this.repository.findOne(
-      { _id: new ObjectId(token.token) },
-      { _id: 0 }
-    );
-      var imageLink = await uploadedFilesToSpaces(image)
-
+      var imageLink = await uploadFile(image)
```
Comment (Bartek-Figat) [inline packages/backEnd/src/services/user.services.ts:155]: Please use **const** instead **var**

File: packages/backEnd/src/validation/index.ts (hunk 4 above)
Comment (BElluu) [reply packages/backEnd/src/validation/index.ts]: deleted, coz not need

**Pull request merged**